pkg/catalog/routes.go

```diff
@@ -347,3 +343,21 @@ func getDefaultWeightedClusterForService(meshService service.MeshService) servic
 		Weight:      constants.ClusterWeightAcceptAll,
 	}
 }
+
+// listTrafficTargetPermutations creates a list of TrafficTargets for each source and destination pair.
+func listTrafficTargetPermutations(srcServiceList []service.MeshService, destServiceList []service.MeshService) []trafficpolicy.TrafficTarget {
+	trafficPolicies := make([]trafficpolicy.TrafficTarget, 0, len(srcServiceList)*len(destServiceList))
+
+	for _, destService := range destServiceList {
+		for _, srcService := range srcServiceList {
+			trafficTarget := trafficpolicy.TrafficTarget{
+				Name:        utils.GetTrafficTargetName(srcService, destService),
```

Comment:
I think we should stick to the old naming convention based on my previous comments